clang/lib/Driver/ToolChains/DragonFly.cpp

```diff
@@ -115,16 +117,15 @@ void dragonfly::Linker::ConstructJob(Compilation &C, const JobAction &JA,
           Args.MakeArgString(getToolChain().GetFilePath("crtbegin.o")));
   }
 
-  Args.addAllArgs(CmdArgs, {options::OPT_L, options::OPT_T_Group});
+  Args.addAllArgs(CmdArgs,
```

Comment:
Drop -Z: ld64 specific linker option that is not in GNU ld or ld.lld

     -Z          Do not search the standard directories when searching for
                 libraries and frameworks.